fabric-renderer-indigo/src/main/java/net/fabricmc/indigo/renderer/mixin/MixinChunkBuilderThingy.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.indigo.renderer.mixin;
+
+import java.util.Random;
+import java.util.Set;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.fabricmc.fabric.api.renderer.v1.model.FabricBakedModel;
+import net.fabricmc.indigo.Indigo;
+import net.fabricmc.indigo.renderer.accessor.AccessChunkRendererRegion;
+import net.fabricmc.indigo.renderer.render.TerrainRenderContext;
+import net.minecraft.block.BlockRenderType;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.client.render.BufferBuilder;
+import net.minecraft.client.render.block.BlockRenderManager;
+import net.minecraft.client.render.chunk.BlockLayeredBufferBuilder;
+import net.minecraft.client.render.chunk.ChunkBatcher;
+import net.minecraft.client.render.chunk.ChunkBatcher.ChunkRenderer;
+import net.minecraft.client.render.chunk.ChunkRendererRegion;
+import net.minecraft.client.render.model.BakedModel;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.BlockRenderView;
+
+/**
+ * Implements the main hooks for terrain rendering. Attempts to tread
+ * lightly. This means we are deliberately stepping over some minor
+ * optimization opportunities.<p>
+ * 
+ * Non-Fabric renderer implementations that are looking to maximize 
+ * performance will likely take a much more aggressive approach. 
+ * For that reason, mod authors who want compatibility with advanced
+ * renderers will do well to steer clear of chunk rebuild hooks unless 
+ * they are creating a renderer.<p>
+ * 
+ * These hooks are intended only for the Fabric default renderer and
+ * aren't expected to be present when a different renderer is being used.
+ * Renderer authors are responsible for creating the hooks they need.
+ * (Though they can use these as an example if they wish.)
+ */
+@Mixin(targets = "net.minecraft.client.render.chunk.ChunkBatcher$ChunkRenderer$class_4578")
+public class MixinChunkBuilderThingy {
```

Comment:
Might be best to give this a propper name before it goes into yarn.

Reply:
I agree with the name you posted In #yarn (ChunkRebuildTask)